go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -177,12 +180,11 @@ func transformProjection(ctx *plancontext.PlanningContext, op *operators.Project
 		case operators.Eval:
 			return e.EExpr
 		case operators.Offset:
-			t := ctx.SemTable.ExprTypes[e.Expr]
-			return &evalengine.Column{
-				Offset:    e.Offset,
-				Type:      t.Type,
-				Collation: collations.TypedCollation{},
+			t, found := ctx.SemTable.ExprTypes[e.Expr]
```

Comment:
hmm... shouldn't we use `ctx.SemTable.TypeForExpr` here? I know the old code wasn't doing it, but I don't see why we should touch the internals in this case

Reply:
Agreed, cleaned this up!